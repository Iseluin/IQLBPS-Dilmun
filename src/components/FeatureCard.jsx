import React from 'react';

const FeatureCard = ({ icon, title, description }) => {
  return (
    <div className='flex flex-col justify-center items-center'>
      <div className="flex justify-center items-center mb-4 w-10 h-10 rounded-full bg-secondary lg:h-12 lg:w-12 dark:bg-primary-900">
        {icon}
      </div>
      <h3 className="mb-2 text-xl font-bold dark:text-white">{title}</h3>
      <p className="text-gray-500 dark:text-gray-400">{description}</p>
    </div>
  );
};

export default FeatureCard;
