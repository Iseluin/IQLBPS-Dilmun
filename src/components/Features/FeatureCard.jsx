import React from 'react';

const FeatureCard = ({ icon, title, description, customClass = '' }) => {
  return (
    <div
      data-testid="feature-card"
      className={`flex flex-col justify-center items-center ${customClass}`}
    >
      <div
        data-testid="feature-card-icon"
        className="flex justify-center items-center mb-4 w-10 h-10 rounded-full bg-secondary lg:h-12 lg:w-12"
      >
        {icon}
      </div>
      <h3
        data-testid="feature-card-title"
        className="mb-2 text-xl font-bold text-primary"
      >
        {title}
      </h3>
      <p
        data-testid="feature-card-paragraph"
        className="text-primary"
      >
        {description}
      </p>
    </div>
  );
};

export default FeatureCard;
