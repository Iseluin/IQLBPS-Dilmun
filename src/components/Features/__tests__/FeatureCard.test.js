import { render, screen } from '@testing-library/react';
import { FiFilter } from 'react-icons/fi';
import FeatureCard from '../FeatureCard';

test('feature card should be render', () => {
  render(
    <FeatureCard
      icon={<FiFilter />}
      title="Filter"
      description="Filter your items to find the right information."
    />
  );
  const FeatureCardElement = screen.getByTestId('feature-card');
  expect(FeatureCardElement).toBeInTheDocument();
});
test('feature card should has an icon of (<FiFilter/>)', () => {
  render(
    <FeatureCard
      icon={<FiFilter />}
      title="Filter"
      description="Filter your items to find the right information."
    />
  );
  const FeatureCardIcon = screen.getByTestId('feature-card-icon');
  expect(FeatureCardIcon).toBeTruthy();
});
test('feature card should has an title of (Filter)', () => {
  render(
    <FeatureCard
      icon={<FiFilter />}
      title="Filter"
      description="Filter your items to find the right information."
    />
  );
  const FeatureCardTitle = screen.getByTestId('feature-card-title');
  expect(FeatureCardTitle).toHaveTextContent('Filter');
});

test('feature card should has an paragraph of (Filter your items to find the right information.)', () => {
  render(
    <FeatureCard
      icon={<FiFilter />}
      title="Filter"
      description="Filter your items to find the right information."
    />
  );
  const FeatureCardParagraph = screen.getByTestId('feature-card-paragraph');
  expect(FeatureCardParagraph).toHaveTextContent(
    'Filter your items to find the right information.'
  );
});
