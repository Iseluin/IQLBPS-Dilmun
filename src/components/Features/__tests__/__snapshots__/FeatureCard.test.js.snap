// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`feature card should be render 1`] = `
<div
  class="flex flex-col justify-center items-center "
  data-testid="feature-card"
>
  <div
    class="flex justify-center items-center mb-4 w-10 h-10 rounded-full bg-secondary lg:h-12 lg:w-12 dark:bg-yellow-400"
    data-testid="feature-card-icon"
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        points="22 3 2 3 10 12.46 10 19 14 21 14 12.46 22 3"
      />
    </svg>
  </div>
  <h3
    class="mb-2 text-xl font-bold text-background dark:text-white"
    data-testid="feature-card-title"
  >
    Filter
  </h3>
  <p
    class="text-background dark:text-gray-400"
    data-testid="feature-card-paragraph"
  >
    Filter your items to find the right information.
  </p>
</div>
`;
